x/ccv/consumer/keeper/keeper.go

```diff
@@ -206,76 +207,81 @@ func (k Keeper) DeletePendingChanges(ctx sdk.Context) {
 	store.Delete(types.PendingChangesKey())
 }
 
-// GetElapsedPacketMaturityTimes returns a slice of already elapsed PacketMaturityTimes, sorted by vscIDs,
-// i.e., the slice contains the IDs of the matured VSCPackets
-func (k Keeper) GetElapsedPacketMaturityTimes(ctx sdk.Context) (maturingVSCPacket []consumertypes.MaturingVSCPacket) {
-	currentTime := uint64(ctx.BlockTime().UnixNano())
+// GetElapsedPacketMaturityTimes returns a slice of already elapsed PacketMaturityTimes, sorted by maturity times,
+// i.e., the slice contains the IDs of the matured VSCPackets.
+func (k Keeper) GetElapsedPacketMaturityTimes(ctx sdk.Context) (maturingVSCPackets []consumertypes.MaturingVSCPacket) {
 	store := ctx.KVStore(k.storeKey)
 	iterator := sdk.KVStorePrefixIterator(store, []byte{types.PacketMaturityTimeBytePrefix})
 
 	defer iterator.Close()
 
 	for ; iterator.Valid(); iterator.Next() {
-		vscId := binary.BigEndian.Uint64(iterator.Key()[1:])
-		maturityTime := binary.BigEndian.Uint64(iterator.Value())
+		var maturingVSCPacket consumertypes.MaturingVSCPacket
+		if err := maturingVSCPacket.Unmarshal(iterator.Value()); err != nil {
+			panic(fmt.Errorf("failed to unmarshal MaturingVSCPacket: %w", err))
+		}
 
 		// If the maturity time is after the current time, then stop the iteration;
```

Comment:
Maybe change the comment to reflect the surrounding code (use "before" in both the comment and the function.

```
// If the current block time is before maturity time then stop the iteration;
// this is possible since the iteration over PacketMaturityTimes is in order
// of maturity times
```